contents/docs/developing-locally.md

```diff
@@ -17,18 +17,30 @@ Then start the instance with `docker-compose`:
 docker-compose -f docker-compose.dev.yml up
 ```
 
+> **It may take several minutes to build the system for the first time**. If you see an error message once your app has launched saying the front-end is not built, please wait for the Yarn dependencies to be installed (the logs will output a `✔ Webpack: Compiled successfully` message once the frontend is ready).
+
 ## Using virtualenv
 
-1. Make sure you have python 3 installed `python3 --version`
-2. Make sure you have [Redis installed](https://redis.io/download) and running. 
+1. Make sure you have Python 3 installed `python3 --version`. [pyenv](https://github.com/pyenv/pyenv) is recommended to manage multiple Python versions and make sure you don't use the system version.
+2. Make sure you have [Redis installed](https://redis.io/download) and running.
 
-    On MacOS, this is done with: `brew install redis && brew services start redis`
-3. Make sure you have [PostgreSQL installed](https://www.postgresql.org/download/) and running.
+    ```bash
+    # macOS (Homebrew)
+    brew install redis && brew services start redis
+    ```
 
-    On MacOS, this is done with: `brew install postgresql && brew services start postgresql`
+3. Make sure you have [PostgreSQL installed](https://www.postgresql.org/download/) and running. You may also try [Postgres.app](https://postgresapp.com/), but remember to follow the instructions to add `psql` to your `$PATH` if you do.
 
+    ```bash
+    # macOS (Homebrew)
+    brew install postgresql && brew services start postgresql
+    ```
  
-4. Create the Database: `createdb posthog`
+4. Create the Postgres database with the command `createdb posthog` on the shell or by using the Postgres interactive terminal:
+    ```
+    psql
```

Comment:
Just wondering about the `psql` command. For me at least I can't run it by itself - it asks me for a database name. Once I connect to one, however, I can then run the command to create the db.

Reply:
> Just wondering about the `psql` command. For me at least I can't run it by itself - it asks me for a database name. Once I connect to one, however, I can then run the command to create the db.

Well that depends on how you installed Postgres. Postgres.app for example creates a database with your current username by default so doing `psql` by itself works. I guess the safe option would be to do `psql -d postgres` to make sure you can connect without regard for how you installed Postgres.

Reply:
Yeah exactly - let's go with the safe option